### PR TITLE
Arp suppression changes in ofnet

### PR DESCRIPTION
**Pull request by DivyaVavili**
This change sends out Proxy ARP response in L2 mode which reduces the number of ARP packets being flooded.

ARP request handling in various scenarios:
- Src and Dest EP known:
  - Proxy ARP if Dest EP is present locally on the host
- Src EP known, Dest EP not known:
  - ARP Request to a router/VM scenario. Reinject ARP request to uplinks
- Src EP not known, Dest EP known:
  - Proxy ARP if Dest EP is present locally on the host
- Src and Dest EP not known:
  - Ignore processing the request

### CODE REVIEW

**Comment by shaleman** [comment]
@DivyaVavili 
Changes look good. Thanks for cleaning up ofnet unit tests..
couple of minor comments:

- Is there a way you can add unit tests for ARP processing by pretending to receive a packet? i.e. call ofnetAgent.PacketRcvd() from the unit test code..

- Vlan changes look good. Can we make the same change for vxlan bridge mode too? Its fine to do this as a separate PR

File: vlanBridge.go [hunk 1]
```diff
@@ -287,3 +314,118 @@ func (vl *VlanBridge) initFgraph() error {
...
+				for _, portNo := range vl.uplinkDb {
+					log.Debugf("Sending to uplink: %+v", portNo)
+					pktOut.AddAction(openflow13.NewActionOutput(portNo))
+				}
+
+				// Send the packet out
+				vl.ofSwitch.Send(pktOut)
+			}
+
+		case protocol.Type_Reply:
```
Comment (abhinandanpb) [inline vlanBridge.go]: @DivyaVavili : just to understand the arp redirection to controller logic added previously seems to be redirecting only arp request (in initFgraph)? Do we still see the reply packets at the controller ?

**Comment by DivyaVavili** [comment]
@shaleman: 
I tried simulating packet flows using PacketRcvd, but looks like it would be easy to do it once the libpkt changes come in.
Also, I'm working on the vxlan changes as well. Will raise a PR for it soon.

File: vlanBridge.go (hunk 1 above)
Comment (DivyaVavili) [inline vlanBridge.go]: @abhinandanpb 
The ARP reply packets do not come to the controller. The processing here though is just for future reference/processing.

**Pull request merged**